web/vibe/web/common.d

```diff
@@ -753,6 +905,47 @@ package ParamResult readFormParamRec(T)(scope HTTPServerRequest req, ref T dst,
 	return ParamResult.ok;
 }
 
+// test new array mechanisms
+unittest {
+	import vibe.http.server;
+	import vibe.inet.url;
+
+	auto req = createTestHTTPServerRequest(URL("http://localhost/route?arr_0=1&arr_2=2&arr_=3"));
+	int[] arr;
+	ParamError err;
+	auto result = req.readFormParamRec(arr, "arr", false, NestedNameStyle.underscore, err);
+	assert(result == ParamResult.ok);
+	assert(arr == [1,3,2]);
+
+	// try with static array
+	int[3] staticarr;
+	result = req.readFormParamRec(staticarr, "arr", false, NestedNameStyle.underscore, err);
+	assert(result == ParamResult.ok);
+	assert(staticarr == [1,3,2]);
+
+	// d style
+	req = createTestHTTPServerRequest(URL("http://localhost/route?arr[2]=1&arr[0]=2&arr[]=3"));
+	result = req.readFormParamRec(arr, "arr", false, NestedNameStyle.d, err);
+	assert(result == ParamResult.ok);
+	assert(arr == [2,3,1]);
+
+	result = req.readFormParamRec(staticarr, "arr", false, NestedNameStyle.d, err);
+	assert(result == ParamResult.ok);
+	assert(staticarr == [2,3,1]);
+
+	// try nested arrays
+	req = createTestHTTPServerRequest(URL("http://localhost/route?arr[2][]=1&arr[0][]=2&arr[1][]=3&arr[0][]=4"));
+	int[][] arr2;
+	result = req.readFormParamRec(arr2, "arr", false, NestedNameStyle.d, err);
+	assert(result == ParamResult.ok);
+	assert(arr2 == [[2,4],[3],[1]]);
+
+	int[][2] staticarr2;
```

Comment:
I modified this to a 2-element array to test that the `arr[2][]` field is ignored (found a bug in the process as well).